hazelcast/src/main/java/com/hazelcast/cache/impl/nearcache/impl/DefaultNearCache.java

```diff
@@ -39,58 +39,26 @@
     protected final NearCacheConfig nearCacheConfig;
     protected final NearCacheContext nearCacheContext;
     protected final SerializationService serializationService;
-    protected NearCacheRecordStore<K, V> nearCacheRecordStore;
-    protected ScheduledFuture expirationTaskFuture;
+    protected final NearCacheRecordStore<K, V> nearCacheRecordStore;
+    protected final ScheduledFuture expirationTaskFuture;
 
-    public DefaultNearCache(String name, NearCacheConfig nearCacheConfig,
-                            NearCacheContext nearCacheContext) {
+    public DefaultNearCache(String name, NearCacheConfig nearCacheConfig, NearCacheContext nearCacheContext) {
```

Comment:
If I try to replace the first constructor with `this(name, nearCacheConfig, nearCacheContext, nearCacheContext.getSerializationService(), createNearCacheRecordStore(nearCacheConfig, nearCacheContext), createExpirationTask());` I get the compile error `cannot reference this before supertype constructor has been called`, so I don't know how to achieve this.

Beside the size of the methods which don't make them good candidates for inlineing, we need to override them in EE. So maybe we could go with `newInstance` methods to achieve this, but I don't know if this is a better approach than having two small constructors.